### PR TITLE
chore: refactor upstream form module

### PR DESCRIPTION
**Pull request by juzhiyuan**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

- [x] make components smaller & clearer
- [ ] added default values (will be resolved in a new PR)

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases? (**TODO**)
- [ ] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by juzhiyuan** [comment]
Hi, I noticed that there have too many changes!

image

This PR only separates the [UpstreamForm](https://github.com/apache/apisix-dashboard/blob/master/web/src/components/Upstream/UpstreamForm.tsx) file (more than 813 lines) into some sub-components.

**Comment by juzhiyuan** [comment]
This PR doesn't change logics, only adjust the structure so we could know more clear about this component. Test is under debugging.

**Comment by membphis** [comment]
this is a big PR, need @LiteSun @liuxiran to confirm first

**Comment by juzhiyuan** [comment]
yes, also needs @imjoey's opinion.

**Comment by imjoey** [comment]
@juzhiyuan the adjustment is OK on me side which makes the code structure much more intuitive and easier for maintenance.

**Comment by juzhiyuan** [comment]
yes! And if we want to use a filed like `active.host`, we just need to create a TSX file `./components/active-check/Host.tsx`, and use it `<ActiveCheck.Host />`. That's more easy to use & read!

**Comment by juzhiyuan** [comment]
I'm honored to have this Component got structure updated :)

File: web/src/components/Upstream/constant.ts [hunk 1]
```diff
@@ -16,49 +16,76 @@
...
     {
       host: '',
       port: 80,
       weight: 1
     }
   ],
-  type: 'roundrobin',
+  retries: 0,
   timeout: {
+    // TODO: > 0
```
Comment (LiteSun) [inline web/src/components/Upstream/constant.ts:30]: Need more details.

File: web/src/components/Upstream/constant.ts [hunk 2]
```diff
@@ -16,49 +16,76 @@
...
     connect: 6,
     send: 6,
     read: 6,
   },
-  retries: 1
+  type: 'roundrobin',
+  checks: {},
+  scheme: "http",
```
Comment (LiteSun) [inline web/src/components/Upstream/constant.ts:37]: should keep same style `""` or  `' '`.

File: web/src/components/Upstream/UpstreamForm.tsx [hunk 3]
```diff
@@ -130,487 +106,110 @@ const UpstreamForm: React.FC<Props> = forwardRef(
...
+          {options.map(({ label, name, component }) => (
+            <div key={label}>
+              <Form.Item label={label} name={name} valuePropName="checked" key={label}>
+                <Switch disabled={readonly} />
+              </Form.Item>
+              <Form.Item shouldUpdate noStyle>
+                {() => {
+                  if (form.getFieldValue(name)) {
+                    if (name.includes("active")) {
+                      // TODO: 避免默认值被覆盖
```
Comment (LiteSun) [inline web/src/components/Upstream/UpstreamForm.tsx:199]: remove chinese please.

**Review by LiteSun — APPROVED** [review]

File: web/src/components/Upstream/constant.ts (hunk 2 above)
Comment (juzhiyuan) [reply web/src/components/Upstream/constant.ts:36]: yes, will be formatted globally in a new PR.

Comment (juzhiyuan) [reply web/src/components/Upstream/constant.ts:36]: https://github.com/apache/apisix-dashboard/issues/1746

**Review by imjoey — APPROVED** [review]
/lgtm

Thanks, @juzhiyuan.

**Review by bzp2010 — APPROVED** [review]

**Comment by juzhiyuan** [comment]
Hi @liuxiran, this PR needs your review :) You could review this one first, after issues you find (maybe) are resolved, I will add features for Upstream.

**Review by Yiyiyimu — APPROVED** [review]

**Pull request merged**